front/components/vaults/VaultResourcesList.tsx

```diff
@@ -171,13 +171,49 @@ export const VaultResourcesList = ({
     );
   }
 
-  const setUpDataSources = vaultDataSourceOrViews.map((dsv) => dsv.connectorId);
-  const unusedDataSources = managedDataSources.filter(
-    (ds) => !setUpDataSources.includes(ds.connectorId)
-  );
+  const saveSelectedDatasources = async (
+    selectedDataSources: VaultSelectedDataSources
+  ) => {
+    const body = {
+      content: selectedDataSources.map((sDs) => ({
```

Comment:
A bit orthogonal to the PR but why do we have this concept of `content` vs creating a data source view? cc @flvndvd

Reply:
It's right next to `members` which we know we don't want :) I thiiiiiiink this PATCH method should just be killed with fire no?

Reply:
iiuc this is going to change indeed: https://dust4ai.slack.com/archives/C050SM8NSPK/p1724233687958259?thread_ts=1724232606.765919&cid=C050SM8NSPK

This PR can still be reviewed, and I'll update the save method when the new routes are updated.